api/osb/context_signature.go

```diff
@@ -0,0 +1,119 @@
+package osb
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+)
+
+const ContextSignaturePluginName = "ContextSignaturePlugin"
+
+type ContextSignaturePlugin struct {
+	CtxPrivateKey string
+	CtxPublicKey  string
+}
+
+func NewCtxSignaturePlugin(publicKey, privateKey string) *ContextSignaturePlugin {
+	return &ContextSignaturePlugin{
+		CtxPrivateKey: privateKey,
+		CtxPublicKey:  publicKey,
+	}
+}
+
+func (s *ContextSignaturePlugin) Name() string {
+	return ContextSignaturePluginName
+}
+
+func (s *ContextSignaturePlugin) Provision(req *web.Request, next web.Handler) (*web.Response, error) {
+	return s.sign(req, next)
+}
+
+func (s *ContextSignaturePlugin) Bind(req *web.Request, next web.Handler) (*web.Response, error) {
+	return s.sign(req, next)
+}
```

Comment:
UpdateService? we pass a context in update calls as well

Reply:
done